packages/eslint-plugin/src/rules/pascal-case-enums.ts

```diff
@@ -0,0 +1,75 @@
+import { pascalCase } from "change-case";
+import { Rule } from "eslint";
+
+export default {
+    meta: {
+        type: "layout",
+        docs: {
+            description: "Enforce PascalCase for enum names",
+        },
+        fixable: "code",
+        messages: {
+            pascalCaseEnumDeclarationMessage: "Enum name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberMessage: "Enum member name should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+            pascalCaseEnumMemberLiteralMessage: "Enum member literal should be PascalCase, is: {{name}}, expected: {{pascalCaseName}}",
+        },
+    },
+    create(context) {
+        return {
+            "TSEnumDeclaration > Identifier": (identifier) => {
+                const name = identifier.name;
+                const pascalCaseName = pascalCase(name);
+
+                if (name !== pascalCaseName) {
+                    context.report({
+                        node: identifier,
+                        messageId: "pascalCaseEnumDeclarationMessage",
+                        data: {
+                            name,
+                            pascalCaseName,
+                        },
+                        fix: function (fixer) {
+                            return fixer.replaceText(identifier, pascalCaseName);
+                        },
+                    });
+                }
+            },
+            "TSEnumMember > Identifier": (identifier) => {
+                const name = identifier.name;
+                const pascalCaseName = pascalCase(name);
+
+                if (name !== pascalCaseName) {
+                    context.report({
+                        node: identifier,
+                        messageId: "pascalCaseEnumMemberMessage",
+                        data: {
+                            name,
+                            pascalCaseName,
+                        },
+                        fix: function (fixer) {
+                            return fixer.replaceText(identifier, pascalCaseName);
+                        },
+                    });
+                }
+            },
+            "TSEnumMember > Literal": (literal) => {
```

Comment:
Same for `literal` https://github.com/vivid-planet/comet/pull/1631/files#r1469279812